crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1095,6 +1105,54 @@ impl World {
         Some(&*column.get_data_ptr().as_ptr().cast::<R>())
     }
 
+    /// Gets a resource to the resource with the id [`ComponentId`] if it exists.
+    /// The returned pointer must not be used to modify the resource, and must not be
+    /// dereferenced after the immutable borrow of the [`World`] ends.
+    ///
+    /// You should always prefer to use the [`World::get_resource`] where possible and only
+    /// use this in cases where the actual types are not known at compile time.
```

Comment:
Should these functions be moved to a separate impl block too?

Reply:
Yes, that would be consistent. I moved them down into a separate block.